app_dart/lib/src/request_handlers/scheduler/batch_backfiller.dart

```diff
@@ -97,6 +100,18 @@ class BatchBackfiller extends RequestHandler {
     }
     // Schedule all builds asynchronously
     await Future.wait<void>(futures);
+
+    // Update tasks status as in progress to avoid duplicate scheduling.
+    try {
```

Comment:
Instead of adding a new field to track separately, you can use map to get it from the backfill tuple

```dart
final List<Task> backfillTasks = backfill.map((Tuple<Target, Task, Commit> tuple) => tuple.second).toList();
```

Reply:
Updated.